guides/examples/Amazon SQS.md

```diff
@@ -1,3 +1,185 @@
 # Amazon SQS
 
-TODO.
\ No newline at end of file
+Amazon Simple Queue Service (SQS) is a highly scalable distributed message
+queuing service provided by Amazon.com. AWS SQS offers two types of message
+queues:
+
+  * Standard
+    * Nearly unlimited throughput
+    * Best-effort ordering
+    * At-least-once delivery
+
+  * FIFO
+    * Limited number of transactions per second (TPS).
+      See AWS SQS documentation for more information on limits.
+    * Order in which messages are sent/received is strictly preserved 
+    * Exactly-once delivery
+  
+Broadway can work seamlessly with both, Standard and FIFO queues.
+
+## Getting Started
+
+In order to use Broadway with SQS, we basically need to:
+
+  1. Create a SQS queue (or use an existing one)
+  1. Configure our Elixir project to use Broadway
+  1. Define your pipeline configuration
+  1. Implement Broadway callbacks, so we can process incoming messages
+  1. Tuning the configuration (Optional)
+
+## Create a SQS queue
+
+Amazon provides a comprehensive [Step-by-step Guide](https://aws.amazon.com/getting-started/tutorials/send-messages-distributed-applications/)
+on creating SQS queues. In case you don't have an AWS account and want to
+test Broadway locally, use can easily install [ElasticMQ](https://github.com/softwaremill/elasticmq),
+which is a message queue system that offers a SQS-compatible query interface.
+
+## Configure the project
+
+In this guide we're going to use [BroadwaySQS](https://github.com/plataformatec/broadway_sqs),
+which is a Broadway SQS Connector provided by [Plataformatec](http://www.plataformatec.com).
+
+### Setting up dependencies
+
+Add `:broadway_sqs` to the list of dependencies in `mix.exs` along the HTTP
+client of your choice (defaults to `:hackney`):
+
+    def deps do
+      [
+        ...
+        {:broadway_sqs, "~> 0.1.0"},
+        {:hackney, "~> 1.9"},
+      ]
+    end
+
+## Define the pipeline configuration
+
+Like any other process-based behaviour, implementing the Broadway
+bahaviour is straightforward. The second argument of
+`Broadway.start_link/2` is the pipeline configuration. Assuming we
+want to consume messages from a queue called `my_queue`. The minimal
+configuration would be:
+
+    defmodule MyBroadway do
+      use Broadway
+      alias BroadwaySQS.{SQSProducer, ExAwsClient}
+
+      def start_link(_opts) do
+        Broadway.start_link(__MODULE__,
+          name: __MODULE__,
+          producers: [
+            default: [
+              module: SQSProducer,
+              arg: [
+                sqs_client: {ExAwsClient, [
+                  queue_name: "my_queue",
+                ]}
+              ]
+            ]
+          ],
+          processors: [default: []],
+          batchers: [default: []]
```

Comment:
I would recommend setting batch_size and batch_timeout here. I know you talk about them later, but I would rather be explicit about those two since the beginning.